src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,163 @@
+import classNames from 'classnames';
+import { Todo } from '../../types';
+import { useEffect, useRef, useState } from 'react';
+import { useTodoContext } from '../../TodoContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { setTodos, inputFocus } = useTodoContext();
+  const { completed, title, id } = todo;
+  const [isEdit, setIsEdit] = useState(false);
+  const [editText, setEditText] = useState(title);
+  const [isEsc, setIsEsc] = useState(false);
+  const itemInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEdit) {
+      itemInput.current?.focus();
+
+      window.addEventListener('keyup', (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          setIsEsc(true);
+        }
+      });
+    }
+  }, [isEdit]);
+
+  const reset = () => {
+    setIsEdit(false);
+  };
+
+  const onDelete = (todoId: number) => {
+    setTodos(prevTodos =>
+      prevTodos.filter(currentTodo => currentTodo.id !== todoId),
+    );
+    inputFocus?.current?.focus();
+  };
+
+  const onToggle = (todoId: number) => {
+    const newTodo = {
+      ...todo,
+      completed: !completed,
+    };
+
+    setTodos(prevTodos =>
+      prevTodos.map(currentTodo =>
+        currentTodo.id === todoId ? newTodo : currentTodo,
+      ),
+    );
+    inputFocus?.current?.focus();
+  };
+
+  const handleEditSubmit = () => {
+    if (editText.length) {
+      if (editText !== title) {
+        const newTodo = {
+          ...todo,
+          title: editText.trim(),
+        };
+
+        if (!isEsc) {
+          setTodos(prevTodos =>
+            prevTodos.map(currentTodo =>
+              currentTodo.id === id ? newTodo : currentTodo,
+            ),
+          );
+          itemInput.current?.blur();
+        } else {
+          setEditText(title);
+          itemInput.current?.blur();
+        }
+      }
+
+      reset();
+    } else {
+      if (!isEsc) {
+        onDelete(id);
+      } else {
+        itemInput.current?.blur();
+        setIsEdit(false);
+        setEditText(title);
+      }
+    }
+
+    inputFocus?.current?.focus();
+
+    if (isEsc) {
+      setIsEsc(false);
+    }
+  };
+
+  useEffect(() => {
+    if (isEdit && isEsc) {
+      handleEditSubmit();
+    }
+  }, [isEdit, isEsc]);
```

Comment:
Fix it 
<img width="956" alt="image" src="https://github.com/user-attachments/assets/dae5c3d8-2e46-4729-99d6-03cd1add94a9" />